batch/test/test_batch.py

```diff
@@ -931,12 +932,9 @@ def test_cant_submit_to_default_with_other_ns_creds(client: BatchClient):
     )
     b = bb.submit()
     status = j.wait()
-    if NAMESPACE == 'default':
-        assert status['state'] == 'Success', str((status, b.debug_info()))
-    else:
```

Comment:
Do we not need this test anymore for not being able to log in?

Reply:
This test was meant to make sure that you couldn't submit jobs to default through various "adversarial" attempts at changing the deploy config. However, with this change to mounting the deploy config as readonly, the job can't `rm` or `mv` it anymore, and that's true for both test and production jobs. So this test is primarily to show that we correctly mount the deploy-config as readonly. The above test where I use the env variable is to show that you can't access default from a test namespace even if you override what is inside the deploy config in your code.